packages/types/src/DidDetails.ts

```diff
@@ -211,3 +211,7 @@ export type DidSignature = {
   keyUri: DidResourceUri
   signature: string
 }
+
+export type JsonEnum<K extends string, V> = K extends any
```

Comment:
This looks just like `Record`. What does it do differently and is it important enough to have a new type for it? Because  it makes things more confusing.

Reply:
ok, I played around with a bit and `Record` would not work here.
The only solution I could come up with is:
```
export type JsonEnum<K extends string, V> = {
  [P in K]?: V
}
```
Which I like more, because it is more straightforward and understandable. Only downside is, that it allows empty objects, but I don't see a way around that. On the other side,`makeJsonEnum` would not be needed anymore,  which currently uses type assertion with `as` and is needed, because it would not be possible to assign to the JsonEnum type without it.

Reply:
AFAIK it's the __only__ way to turn `'a' | 'b'` into something like `{a: any} | {b: any}`, which is the sort of type that polkadot requires for the creation of enums. What you proposed gives you `{a?: any, b?: any}` which besides empty objects also allows `{a: 1, b: 2}`. And it is not assignable to the polkadot type, so I don't think that will work.

Reply:
Another way is to list the types explicitly: `{a: any} | {b: any}`. It is wordier and longer (a downside) but also explicit (an upside).

Reply:
but what does that solve? if variable `a` is typed as a union of string literals, typescript interprets `{[a]: any}` as `{[key: string]: any}` and will not accept it for a signature that requires `{a: any} | {b: any}`